core/variant/type_info.h

```diff
@@ -367,4 +367,75 @@ ZERO_INITIALIZER_NUMBER(char32_t)
 ZERO_INITIALIZER_NUMBER(float)
 ZERO_INITIALIZER_NUMBER(double)
 
+template <typename T>
+inline constexpr bool is_variant_type_v = TypeInherits<Object, T>::value;
+
+template <typename T, typename = void>
+struct GetVariantType;
+
+template <typename T>
+struct GetVariantType<T, typename EnableIf<TypeInherits<Object, T>::value>::type> {
+	static const Variant::Type value = Variant::OBJECT;
+};
+
+template <typename T>
+inline constexpr Variant::Type get_variant_type_v = GetVariantType<T>::value;
```

Comment:
Not sure it helps that much to declare these `constexpr` values. I think `GetVariantType<T>::value` is just as readable and easy to type as `get_variant_type_v`

Reply:
I assumed that this was required to make the `if constexpr` logic work, but after testing that doesn't appear to be the case! However, I did end up brainstorming an implementation that doesn't need the struct setup in the first place, so the `constexpr` values will stay because they can now exist in isolation